test/abstract_syntax_tree.coffee

```diff
@@ -4,37 +4,52 @@
 # Recursively compare all values of enumerable properties of `expected` with
 # those of `actual`. Use `looseArray` helper function to skip array length
 # comparison.
-deepStrictEqualExpectedProperties = (actual, expected) ->
-  white = (text, values...) -> (text[i] + "#{reset}#{value}#{red}" for value, i in values).join('') + text[i]
+deepStrictIncludeExpectedProperties = (actual, expected) ->
   eq actual.length, expected.length if expected instanceof Array and not expected.loose
   for key, val of expected
     if 'object' is typeof val
-      fail white"Property #{key} expected, but was missing" unless actual[key]
-      deepStrictEqualExpectedProperties actual[key], val
+      fail "Property #{reset}#{key}#{red} expected, but was missing" unless actual[key]
+      deepStrictIncludeExpectedProperties actual[key], val
     else
-      eq actual[key], val, white"Property #{key}: expected #{actual[key]} to equal #{val}"
+      eq actual[key], val, """
+        Property #{reset}#{key}#{red}: expected #{reset}#{actual[key]}#{red} to equal #{reset}#{val}#{red}
+          Expected AST output to include:
+          #{reset}#{inspect expected, {depth: 10, colors: yes}}#{red}
+          but instead it was:
+          #{reset}#{inspect actual, {depth: 10, colors: yes}}#{red}
+      """
   actual
 
+# Flag array for loose comparison. See reference to `.loose` in
+# `deepStrictIncludeExpectedProperties` above.
+looseArray = (arr) ->
+  Object.defineProperty arr, 'loose',
+    value: yes
+    enumerable: no
+  arr
+
+# Helpers to get AST nodes for a string of code. The root node is always a
+# `Block` node, so for brevity in the tests return its children from
+# `expressions`.
+getAstExpressions = (code) ->
+  ast = CoffeeScript.compile code, ast: yes
+  ast.expressions
+
+getAstExpression = (code) -> getAstExpressions(code)[0]
+
 testExpression = (code, expected) ->
-  ast = getExpressionAst code
+  ast = getAstExpression code
   if expected?
-    deepStrictEqualExpectedProperties ast, expected
+    deepStrictIncludeExpectedProperties ast, expected
   else
     # Convenience for creating new tests; call `testExpression` with no second
     # parameter to see what the current AST generation is for your input code.
-    console.log require('util').inspect ast,
+    console.log inspect ast,
```

Comment:
I don't think your `inspect` "polyfill" would work here, wouldn't you be calling `console.log(console.log(whatever))`?